.github/workflows/reusable-e2e.yml

```diff
@@ -301,6 +301,295 @@ jobs:
+            **/node_modules
+          key: ${{ runner.os }}-nodemodules-android-${{ inputs.commit }}
+        env:
+          SEGMENT_DOWNLOAD_TIMEOUT_MINS: 3
+
+      - name: Restore ui/dist cache
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1 https://github.com/actions/cache/commit/88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8
+        id: restore-ui-cache
+        with:
+          path: ./packages/ui/dist
+          key: ${{ runner.os }}-ui-android-${{ inputs.commit }}
+
+      - name: Restore react-native/dist cache
+        id: restore-package-cache
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1 https://github.com/actions/cache/commit/88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8
+        with:
+          path: ./packages/react-native/dist
+          key: ${{ runner.os }}-react-native-android-${{ inputs.commit }}
+        env:
+          SEGMENT_DOWNLOAD_TIMEOUT_MINS: 1
+
+      - name: Install packages
+        if: steps.restore-cache.outputs.cache-hit != 'true'
+        uses: ./.github/actions/install-with-retries
+        with:
+          skip-cypress-binary: true
+
+      - name: Build @aws-amplify/ui package
+        if: steps.restore-ui-cache.outputs.cache-hit != 'true'
+        run: yarn ui build
+
+      - name: Build react-core package
+        if: steps.restore-package-cache.outputs.cache-hit != 'true'
+        run: yarn react-core build
+
+      - name: Build react-native package
+        if: steps.restore-package-cache.outputs.cache-hit != 'true'
+        run: yarn react-native build
+
+      - name: Add Amplify CLI
+        run: yarn global add @aws-amplify/cli
+
+      - name: Get CLI versions
+        id: cli-version
+        run: echo "version=$(amplify --version)" >> $GITHUB_OUTPUT
+
+      - name: Configure auth credentials
+        uses: aws-actions/configure-aws-credentials@e1e17a757e536f70e52b5a12b2e8d1d1c60e04ef # v2.0.0 https://github.com/aws-actions/configure-aws-credentials/commit/e1e17a757e536f70e52b5a12b2e8d1d1c60e04ef
+        with:
+          aws-region: us-east-2
+          role-to-assume: ${{ secrets.AUTH_E2E_ROLE_ARN }}
+
+      # Amplify CLI does not support headless pull with temporary credentials
+      # when useProfile is false.
+      # See: https://github.com/aws-amplify/amplify-cli/issues/11009.
+      - name: Create temp AWS Profile
+        run: |
+          aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID && \
+          aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY && \
+          aws configure set aws_session_token $AWS_SESSION_TOKEN && \
+          aws configure set default.region $AWS_REGION
+
+      - name: Pull down Auth AWS environments
+        run: yarn environments auth pull
+
+      - name: Delete AWS profile
+        run: rm -rf ~/.aws
+
+      - name: Install emulator
+        run: |
+          echo "y" | $ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager --install "system-images;android-27;default;x86_64"
```

Comment:
Emulator fails to launch on API 31, details in github actions [issue](https://github.com/actions/runner-images/issues/6369#issuecomment-1275854074). For API >= 28 Detox needs additional config to enable unencrypted traffic ([docs](https://wix.github.io/Detox/docs/introduction/project-setup#43-enabling-unencrypted-traffic-for-detox)), for this reason picked API 27.

Reply:
A bit concerned with using API 27 to run our tests against as it was released over 5 years ago and last received a security update in October of 2021

Reply:
Yep, I have one!